cranelift/codegen/Cargo.toml

```diff
@@ -18,8 +18,7 @@ bumpalo = "3"
 cranelift-codegen-shared = { path = "./shared", version = "0.92.0" }
 cranelift-entity = { workspace = true }
 cranelift-bforest = { workspace = true }
-cranelift-egraph = { workspace = true }
-hashbrown = { workspace = true, optional = true }
+hashbrown = { workspace = true }
```

Comment:
What exactly is hashbrown necessary for? std::collection::HashMap is a wrapper around hashbrown.

Reply:
It's necessary for `crate::ctxhash::CtxHashMap`, which is a hashmap that uses external context for hashing and equality. `HashMap` isn't sufficient for this.

Note that this was previously in the `cranelift-egraph` crate, which was unconditionally required by `cranelift-codegen`, and now has been moved into `cranelift-codegen`; so there's no change in the dependency footprint.